dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/vitals/MemoryVitalReaderTest.kt

```diff
@@ -0,0 +1,131 @@
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.junit.jupiter.api.io.TempDir
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class MemoryVitalReaderTest {
+
+    lateinit var testedReader: VitalReader
+
+    @TempDir
+    lateinit var tempDir: File
+
+    lateinit var fakeFile: File
+
+    @StringForgery(regex = "(\\.[a-z]+)+")
+    lateinit var fakeName: String
+
+    @StringForgery(regex = "[RSDZTtWXxKWP]")
+    lateinit var fakeState: String
+
+    @IntForgery(1)
+    var fakePid: Int = 0
+
+    @IntForgery(1, 0x7F)
+    var fakeVmRss: Int = 0
+
+    @IntForgery(1, 256)
+    var fakeThreads: Int = 0
+
+    lateinit var fakeStatusContent: String
+
+    @BeforeEach
+    fun `set up`() {
+        fakeFile = File(tempDir, "stat")
+        fakeStatusContent = generateStatusContent()
+        testedReader = MemoryVitalReader(fakeFile)
+    }
+
+    @Test
+    fun `𝕄 read correct data 𝕎 readVitalData()`() {
+        // Given
+        fakeFile.writeText(fakeStatusContent)
+
+        // When
+        val result = testedReader.readVitalData()
+
+        // Then
+        Assertions.assertThat(result).isEqualTo(fakeVmRss.toDouble())
+    }
+
+    @Test
+    fun `𝕄 read correct data 𝕎 readVitalData() {multiple times}`(
+        @IntForgery(1) vmRssValues: List<Int>
+    ) {
+        // Given
+        val results = mutableListOf<Double>()
+        // When
+        vmRssValues.forEach { vmRss ->
+            fakeVmRss = vmRss
+            fakeFile.writeText(generateStatusContent())
+            val result = testedReader.readVitalData()
+            results.add(result!!)
+        }
+
+        // Then
+        Assertions.assertThat(results).isEqualTo(vmRssValues.map { vmRss -> vmRss.toDouble() })
+    }
+
+    @Test
+    fun `𝕄 return null 𝕎 readVitalData() {file doesn't exist}`() {
+        // When
+        val result = testedReader.readVitalData()
+
+        // Then
+        Assertions.assertThat(result).isNull()
+    }
+
+    @Test
+    fun `𝕄 return null 𝕎 readVitalData() {file isn't readable}`() {
+        // Given
+        val restrictedFile = mock<File>()
+        whenever(restrictedFile.exists()) doReturn true
+        whenever(restrictedFile.canRead()) doReturn false
+
+        // When
+        val result = testedReader.readVitalData()
+
+        // Then
+        Assertions.assertThat(result).isNull()
+    }
+
+    private fun generateStatusContent(): String {
+        return mapOf<String, Any>(
+            "Name" to fakeName,
+            "State" to fakeState,
+            "Pid" to fakePid,
+            "VmRSS" to "$fakeVmRss kB".padStart(11, ' '),
```

Comment:
why do we pad here? to check that we use `s` token in the regex to cover any whitespace character?

Reply:
This is because this value will be padded in the real file, so we need to make sure it's handled properly when parsing the line.